zendesk/ticket.go

```diff
@@ -9,33 +9,64 @@ import (
 	"time"
 )
 
+type CustomField struct {
+	ID int64 `json:"id"`
+	// Valid types are string or []string.
+	Value interface{} `json:"value"`
+}
+
+// Custom Unmarshal function required because a custom field's value can be
+// a string or array of strings.
+func (cf *CustomField) UnmarshalJSON(data []byte) error {
+	var temp map[string]interface{}
+	if err := json.Unmarshal(data, &temp); err != nil {
+		return err
+	}
+
+	cf.ID = int64(temp["id"].(float64))
+
+	switch v := temp["value"].(type) {
+	case string:
+		cf.Value = v
+	case []interface{}:
+		var list []string
+		for _, v := range temp["value"].([]interface{}) {
+			list = append(list, v.(string))
```

Comment:
i believe this could panic if the value was something like a list of integers. 

- Might be worth adding a unit test for that.
- probably should check that this assertion is ok and if not return an error.

Reply:
Added unit test for it.

Also checked type during unmarshall:
https://github.com/nukosuke/go-zendesk/blob/d2f149283d372f1748ccffd7deae4f5e699bdca0/zendesk/ticket.go#L34-L40